### PR TITLE
TimeLock Agent Part 5 - Make TimeLockAgent Concrete

### PR DESCRIPTION
**Pull request by jeremyk-91**
**Goals (and why)**: Since we only support the Paxos impl, it doesn't seem necessary to maintain a generic `TimeLockAgent`.

**Implementation Description (bullets)**:
- Is the assertion under "Goals" reasonable?

**Concerns (what feedback would you like?)**:
- We might need to reverse this in the future if we support Raft. But I think we can take on the complexity as it is now.
- Are there any other landmines?

**Where should we start reviewing?**: `TimeLockAgent.java`

**Priority (whenever / two weeks / yesterday)**: soon, blocks #2247

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/2251)

### CODE REVIEW

File: timelock-agent/src/main/java/com/palantir/timelock/paxos/TimeLockAgent.java [hunk 1]
```diff
@@ -16,85 +16,109 @@
...
         registerPaxosResource();
-        registerPaxosExceptionMappers();
+        registerExceptionMappers();
         leadershipCreator.registerLeaderElectionService();
 
         // Finally, register the endpoints associated with the clients.
-        super.createAndRegisterResources();
+        registerTimeLockResourcesForClients();
+
+        ClockSkewMonitorCreator.create(install, registrar).registerClockServices();
```
Comment (tpetracca) [inline timelock-agent/src/main/java/com/palantir/timelock/paxos/TimeLockAgent.java]: Is a separate conversation, but I'd be curious in understanding how this works? I assume it spins of some background thread that checks for clock skew? Might want to embed a way into the Agent to allow for wiring this up to your web servers executor service or w/e

Comment (nziebart) [inline timelock-agent/src/main/java/com/palantir/timelock/paxos/TimeLockAgent.java]: yeah, this uses a single daemon thread to check for skew. ideally we'd use a managed executor service, but we have a pattern all over of just using a daemon thread executor that has been working so far...

File: timelock-agent/src/main/java/com/palantir/timelock/paxos/TimeLockAgent.java [hunk 2]
```diff
@@ -16,85 +16,109 @@
...
 
-        this.paxosResource = PaxosResource.create(paxosInstall.dataDirectory().toString());
+        this.paxosResource = PaxosResource.create(install.paxos().dataDirectory().toString());
         this.leadershipCreator = new PaxosLeadershipCreator(install, runtime, registrar);
         this.lockCreator = new LockCreator(runtime, deprecated);
         this.timestampCreator = new PaxosTimestampCreator(paxosResource,
                 PaxosRemotingUtils.getRemoteServerPaths(install),
                 PaxosRemotingUtils.getSslConfigurationOptional(install).map(SslSocketFactories::createSslSocketFactory),
-                runtime.map(PaxosAgent::getAlgorithmConfigOrDefault));
+                runtime.map(TimeLockAgent::getAlgorithmConfigOrDefault));
```
Comment (nziebart) [inline timelock-agent/src/main/java/com/palantir/timelock/paxos/TimeLockAgent.java]: should this method get renamed / removed?

Comment (jeremyk-91) [inline timelock-agent/src/main/java/com/palantir/timelock/paxos/TimeLockAgent.java]: Mm, good spot. Yeah, it should be. I'll check whether we should _rename_ or _remove_ it and update here

**Pull request merged**